### PR TITLE
Update CI integration and bugfixing

### PR DESCRIPTION
**Pull request by aleksey-kotsuba**
- Developer(s): @aleksey-kotsuba

- Peer Reviewer:

- Tickets:
        https://spryker.atlassian.net/browse/ECO-1014 
        https://spryker.atlassian.net/browse/ECO-1049
        https://spryker.atlassian.net/browse/ECO-1062 
        https://spryker.atlassian.net/browse/ECO-1075 
        https://spryker.atlassian.net/browse/ECO-1164
        https://spryker.atlassian.net/browse/ECO-1202
        https://spryker.atlassian.net/browse/ECO-1203
        https://spryker.atlassian.net/browse/ECO-1204
        https://spryker.atlassian.net/browse/ECO-1206
        https://spryker.atlassian.net/browse/ECO-1240
        https://spryker.atlassian.net/browse/ECO-1241
        https://spryker.atlassian.net/browse/ECO-1248
        https://spryker.atlassian.net/browse/ECO-1253
        https://spryker.atlassian.net/browse/ECO-1255
        https://spryker.atlassian.net/browse/ECO-1299
        https://spryker.atlassian.net/browse/ECO-1300

- Suite Nonsplit PR:

- Demoshop Nonsplit PR:

- Academy PR: https://github.com/spryker/madcap.spryker.github.io/pull/298

#### Please confirm

- [ ] No new OS components - or they have been approved by legal department
- [ ] All new or heavily changed classes get an "A" rating (Scrutinizer), except Facades, Factories, QueryContainers and Tests

#### Documentation

- [ ] Functional documentation provided or in progress?
- [ ] Integration guide for projects provided or not needed?
- [ ] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   Payone                | patch                 |                       |

#### Release Notes

-----------------------------------------

#### Module Payone

_**Patch:** Backwards-compatible bug fix_

Def of done (by responsible developer):
- [ ] All changes are backward-compatible. Outdated code is marked as deprecated.
- [ ] The change is isolated and not mixed with any cleanups or other changes.
- [ ] New code fits to the architecture rules.
- [ ] All new or changed business logic is covered by functional tests (in Zed business layer).

### Bugfixes
 - CI integration has been changed to using ecoci repo.
 - Missed dependencies has been added.
 - Updated module to compatibility with latest demoshop (symfony3.0 included).
 - Fixed several bugs in Express Checkout (integration with shop has been updated).

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "spryker/kernel": "^3.0.0",
     "spryker/oms": "^6.0.0 || ^7.0.0 || ^8.0.0",
     "spryker/propel-orm": "^1.0.0",
+    "spryker/step-engine": "^3.0.0",
```
Comment (profuel) [inline composer.json]: But it's not used in the code. Why you need it here?

File: src/SprykerEco/Zed/Payone/Communication/Controller/TransactionController.php [hunk 2]
```diff
@@ -62,42 +61,38 @@ public function statusUpdateAction(Request $request)
...
      */
-    protected function triggerEventsOnSuccess(TransactionStatusResponse $response, $transactionId, array $dataArray)
+    protected function triggerEventsOnSuccess(PayoneTransactionStatusUpdateTransfer $payoneTransactionStatusUpdateTransfer)
     {
-        if (!$response->isSuccess()) {
+        if (!$payoneTransactionStatusUpdateTransfer->getIsSuccess()) {
             return;
         }
 
         //TODO: Refactor as per CD-380
```
Comment (profuel) [inline src/SprykerEco/Zed/Payone/Communication/Controller/TransactionController.php]: What about this TODO ?

Comment (dereuromark) [reply src/SprykerEco/Zed/Payone/Communication/Controller/TransactionController.php]: Should be removed and can be added to https://release.spryker.com/module-cleanups

File: composer.json (hunk 1 above)
Comment (dereuromark) [reply composer.json]: I see a few

    use Spryker\Yves\StepEngine\Dependency\Form\SubFormInterface;

etc

Comment (darius-telycenas) [reply composer.json]: as far as I remember, without this `interface` the _payment_ step will render a quite broken layout for the form…

Comment (profuel) [reply composer.json]: :+1:

File: src/SprykerEco/Yves/Payone/Form/OnlineTransferSubForm.php [hunk 3]
```diff
@@ -52,21 +51,29 @@ public function getTemplatePath()
...
      */
-    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    public function configureOptions(OptionsResolver $resolver)
     {
-        parent::setDefaultOptions($resolver);
-
         $resolver->setDefaults([
             'data_class' => PayonePaymentOnlinetransferTransfer::class,
             'constraints' => [
                 // Add Callback constraint for bank account check in ancestor classes
```
Comment (profuel) [inline src/SprykerEco/Yves/Payone/Form/OnlineTransferSubForm.php]: What about this part?

File: src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php [hunk 4]
```diff
@@ -158,7 +170,7 @@ protected function hydrateQuoteWithCustomer(
             $quoteTransfer->setCustomer($customerTransfer);
             return $quoteTransfer;
         }
-
+        $customerTransfer->setIsGuest(true);
```
Comment (profuel) [inline src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php]: isn't this set by default checkout process?

File: src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php [hunk 5]
```diff
@@ -178,20 +190,25 @@ protected function hydrateQuoteWithAddresses(
...
+        $address->setFirstName($details->getShippingFirstName());
+        $address->setLastName($details->getShippingLastName());
+        $address->setCompany($details->getShippingCompany());
+        $address->setAddress1($details->getShippingStreet());
+        $address->setAddress2($details->getShippingAddressAdition());
+        $address->setCity($details->getShippingCity());
+        $address->setState($details->getShippingState());
+        $address->setIso2Code($details->getShippingCountry());
+        $address->setZipCode($details->getShippingZip());
+        $quoteTransfer->setBillingAddress($address);
```
Comment (profuel) [inline src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php]: as far as I remember, in Paypal it's possible to have different shipping & billing addresses.

File: src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/AbstractMapper.php [hunk 6]
```diff
@@ -149,10 +151,10 @@ protected function mapBillingAddressToPersonalContainer(PersonalContainer $perso
         $personalContainer->setAddressAddition($billingAddressEntity->getAddress3());
         $personalContainer->setZip($billingAddressEntity->getZipCode());
         $personalContainer->setCity($billingAddressEntity->getCity());
-        $personalContainer->setState($billingAddressEntity->getRegion());
         $personalContainer->setEmail($billingAddressEntity->getEmail());
         $personalContainer->setTelephoneNumber($billingAddressEntity->getPhone());
         $personalContainer->setLanguage($this->getStandardParameter()->getLanguage());
+        $personalContainer->setPersonalid($orderEntity->getCustomerReference());
```
Comment (profuel) [inline src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/AbstractMapper.php]: why not `setPersonalId` ?

File: src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/CreditCardPseudo.php [hunk 7]
```diff
@@ -95,10 +95,10 @@ protected function mapPaymentToAbstractAuthorization(SpyPaymentPayone $paymentEn
         $authorizationContainer->setCurrency($this->getStandardParameter()->getCurrency());
         $authorizationContainer->setPaymentMethod($this->createPaymentMethodContainerFromPayment($paymentEntity));
 
-        $billingAddressEntity = $paymentEntity->getSpySalesOrder()->getBillingAddress();
+        $orderEntity = $paymentEntity->getSpySalesOrder();
```
Comment (profuel) [inline src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/CreditCardPseudo.php]: This is bad practice to work with DB entities. Please try to work only with transfer objects

Comment (profuel) [reply src/SprykerEco/Zed/Payone/Business/Payment/MethodMapper/CreditCardPseudo.php]: Also, it looks like orderEntity is used only by `mapBillingAddressToPersonalContainer` function.
In this case it's suggested to pass `paymentEntity` into function, and let it decide which data is required to make it work.

File: src/SprykerEco/Yves/Payone/Form/OnlineTransferSubForm.php (hunk 3 above)
Comment (aleksey-kotsuba) [reply src/SprykerEco/Yves/Payone/Form/OnlineTransferSubForm.php]: As I understood it came from core6 have no idea what to do here.

File: src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php (hunk 4 above)
Comment (aleksey-kotsuba) [reply src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php]: No, it's not. I checked it. Without this flag customer created after placing order.

File: src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php (hunk 5 above)
Comment (aleksey-kotsuba) [reply src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php]: When I was placing order there was only ship to address at least during express checkout.

Comment (profuel) [reply src/SprykerEco/Yves/Payone/Handler/ExpressCheckout/QuoteHydrator.php]: that's strange and true.
If you don't have requirements from PayPal about implementation, then keep it like this.

**Pull request merged**